types/wlr_surface.c

```diff
@@ -856,7 +878,20 @@ struct wlr_subsurface *wlr_surface_make_subsurface(struct wlr_surface *surface,
 	}
 	subsurface->synchronized = true;
 	subsurface->surface = surface;
+	subsurface->resource =
+		wl_resource_create(client, &wl_subsurface_interface, version, id);
```

Comment:
Should these have an upper bound?
The implementation is for surfaces of version 1, which would not work with higher version surfaces (libwayland-client aborts when the listener has a NULL handler, not sure about libwayland-server).

Reply:
These interfaces are a special case since their constructor is public. I'm still not sure we should keep these public tbh. But yeah, we should add version checks, these would be useful if the compositor decides to use the constructor on his own without `wlr_subcompositor` and supports a more recent version of wl-subcompositor.